### PR TITLE
adding hdu index in DatabaseEntry

### PR DESCRIPTION
**Pull request by abit2**
issue https://github.com/sunpy/sunpy/issues/1771 
ping @Cadair

### CODE REVIEW

**Comment by abit2** [comment]
@Cadair can you take a look

**Comment by Cadair** [comment]
thanks @abit2 the approach looks good. Can you add tests and a changelog entry please?

**Comment by Cadair** [comment]
also ping @Alex-Ian-Hamilton you might find this useful.

File: sunpy/database/tables.py [hunk 1]
```diff
@@ -620,7 +622,7 @@ def _create_display_table(database_entries, columns=None, sort=False):
 
     """
     if columns is None:
-        columns = ['id', 'observation_time_start', 'observation_time_end',
+        columns = ['id','hdu_index' 'observation_time_start', 'observation_time_end',
```
Comment (Alex-Ian-Hamilton) [inline sunpy/database/tables.py]: Missing a comma.

File: sunpy/database/tests/test_database.py [hunk 2]
```diff
@@ -501,6 +501,15 @@ def test_add_from_file(database):
     for entry in database:
         assert entry.fileid == fileid
 
+def test_add_from_file_hdu_index(database):
+    assert len(database) == 0
+    database.add_from_file(RHESSI_IMAGE)
+    assert len(database) == 4
+    # make sure that all entries have the same fileid
+    i = 0
+    for entry in database:
```
Comment (Cadair) [inline sunpy/database/tests/test_database.py]: for this you should use `enumerate`:

```python
for i, entry in enumerate(database):
    assert entry.hdu_index == i
```

**Comment by abit2** [comment]
@Cadair why are there so many errors in travisCI not able to undestand

**Comment by Cadair** [comment]
the build is messed up, it's not your fault #1953 is my attempt to fix them but I can't work it out at the moment.

**Comment by Cadair** [comment]
@sunpy/sunpy-maintainers review please

**Pull request merged**